### PR TITLE
feat: aggregating multiday metadata and patch display info before api return

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/click-stream-api.ts [hunk 1]
```diff
@@ -173,6 +173,17 @@ export class ClickStreamApiConstruct extends Construct {
       },
     });
 
+    const analyticsDisplayTable = new Table(this, 'ClickstreamAnalyticsDisplay', {
```
Comment (zxkane) [inline src/control-plane/backend/click-stream-api.ts:176]: Could we share the table Meta?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/click-stream-api.ts (hunk 1 above)
Comment (tyyzqmf) [reply src/control-plane/backend/click-stream-api.ts]: moved display data to metadata table

File: frontend/public/locales/zh/analytics.json [hunk 2]
```diff
@@ -126,6 +127,10 @@
         "tableLoading": "加载中",
```
Comment (chenhaiyun) [inline frontend/public/locales/zh/analytics.json:127]: Duplicated with line 106 and 107?

File: frontend/src/pages/analytics/metadata/table/MetadataDictionaryTable.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        const { success, message }: ApiResponse<null> =
+          await updateMetadataDisplay({
+            id: `DICTIONARY#${parameter.id}#${attribute.value}`,
+            projectId: parameter.projectId,
+            appId: parameter.appId,
+            displayName: attribute.displayValue,
+            description: '',
+          });
+        if (!success) {
+          throw new Error(message);
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/metadata/table/MetadataDictionaryTable.tsx:186]: Change to `alertMsg`?

File: frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributeSplitPanel.tsx [hunk 4]
```diff
@@ -71,9 +71,13 @@ const MetadataUserAttributeSplitPanel: React.FC<
     }
 
     try {
-      const { success }: ApiResponse<null> = await updateMetadataUserAttribute(
-        attributeDetails
-      );
+      const { success }: ApiResponse<null> = await updateMetadataDisplay({
+        id: `USER_ATTRIBUTE#${attributeDetails.id}`,
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributeSplitPanel.tsx:75]: move `USER_ATTRIBUTE`  as a constant

File: frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributes.tsx [hunk 5]
```diff
@@ -228,7 +228,13 @@ const MetadataUserAttributes: React.FC = () => {
   ) => {
     try {
       const { success, message }: ApiResponse<null> =
-        await updateMetadataUserAttribute(newItem as IMetadataUserAttribute);
+        await updateMetadataDisplay({
+          id: `USER_ATTRIBUTE#${newItem.id}`,
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributes.tsx:232]: `USER_ATTRIBUTE` as a constant

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 6]
```diff
@@ -270,6 +270,7 @@ const AnalyticsPath: React.FC = () => {
       console.log(error);
       return [];
     }
+    return [];
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:273]: Why add this line?

Comment (tyyzqmf) [reply frontend/src/pages/analytics/path/AnalyticsPath.tsx:273]: removed

**Review by zxkane — APPROVED** [review]

**Pull request merged**